### PR TITLE
feat: `drop table .. all`

### LINKED ISSUE #5613: drop table <table_name> all
**Summary**

Now, if we want to drop data as dropping a table, we should:
```
TRUNCATE TABLE <table_name>;
OPTIMIZE TABLE <table_name> PURGE;
DROP TABLE <table_name>;
```
https://databend.rs/doc/reference/sql/ddl/table/ddl-drop-table

For simplicity, we can combine the above commands into one, like:
```
drop table <table_name> all;
```

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

`drop table .. all` **Legacy Parser ONLY**

Besides removing table metadata from KV server,  with the option `all`, the stmt will truncate table data and purge them as well.

## Changelog

- Improvement
- Documentation
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #5613

### CODE REVIEW

**Comment by BohuTANG** [comment]
Cool.
It seems we should add a stateless test for it, then it will check by new planner.

**Comment by dantengsky** [comment]
> Cool. It seems we should add a stateless test for it, then it will check by new planner.

- `drop ... all` impled in new parser, and new `Statement` modified accordingly.
- simple test cases of `drop .. all` added in stateless test
  just to ensure that the new parsers can parse it
  since it is not easy to test the functionality of `drop ... all`, since we have check the underlying storage
-  functionalities is `drop all` is checked in the previous commits :
  https://github.com/datafuselabs/databend/blob/19d51e66e431c5e8f54443d0b697c472a8d65b1e/query/tests/it/storages/fuse/operations/purge_drop.rs#L41-L62

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
Seems this PR break the compat test:
```
The failure tests:
    /workspace/tests/suites/0_stateless/05_ddl/05_0000_ddl_drop_tables.sql

--- /workspace/tests/suites/0_stateless/05_ddl/05_0000_ddl_drop_tables.result	2022-05-26 23:47:40.000806565 +0000
+++ /workspace/tests/suites/0_stateless/05_ddl/05_0000_ddl_drop_tables.stdout	2022-05-26 23:50:47.354907500 +0000
@@ -0,0 +1,3 @@
+ERROR 1105 (HY000) at line 12: Code: 1005, displayText = sql parser error: Expected end of statement, found: ALL.
+ERROR 1105 (HY000) at line 14: Code: 2302, displayText = Table 't' already exists.
+ERROR 1105 (HY000) at line 15: Code: 1005, displayText = sql parser error: Expected end of statement, found: ALL.
```

How to deal with this case:
The test file is for the new version query, and the old-version query to do the compact test? cc @drmingdrmer

**Comment by dantengsky** [comment]
> Seems this PR break the compat test:

those cases are moved into 

[13_ddl_incompatible/13_0001_ddl_drop_table_full.sql](https://github.com/datafuselabs/databend/pull/5619/files#diff-d5f2ac78c2f72fdca81a3d4fc08be5ade8770fd0893bb2802065e2db17c13a9c)

File: query/src/interpreters/interpreter_table_drop.rs [hunk 1]
```diff
@@ -77,6 +78,21 @@ impl Interpreter for DropTableInterpreter {
...
+            if self.plan.all {
+                // errors of truncation are ignored
+                let _ = tbl
+                    .truncate(self.ctx.clone(), TruncateTablePlan {
+                        catalog: self.plan.catalog.clone(),
+                        db: self.plan.db.clone(),
+                        table: self.plan.table.clone(),
+                        purge: true,
+                    })
+                    .await;
```
Comment (drmingdrmer) [inline query/src/interpreters/interpreter_table_drop.rs:92]: Is there a consistency issue that `drop_table` at line 79 and `truncate()` at line 86 will take place on different tables?

Comment (dantengsky) [reply query/src/interpreters/interpreter_table_drop.rs:92]: yeah, there is.  seem we need something like `catalog.drop_table_by_id` here...

**Comment by BohuTANG** [comment]
Let's merge first.

File: query/src/interpreters/interpreter_table_drop.rs (hunk 1 above)
Comment (BohuTANG) [reply query/src/interpreters/interpreter_table_drop.rs:92]: https://github.com/datafuselabs/databend/issues/5626

**Comment by drmingdrmer** [comment]
> Seems this PR break the compat test:
> 
> ```
> The failure tests:
>     /workspace/tests/suites/0_stateless/05_ddl/05_0000_ddl_drop_tables.sql
> 
> --- /workspace/tests/suites/0_stateless/05_ddl/05_0000_ddl_drop_tables.result	2022-05-26 23:47:40.000806565 +0000
> +++ /workspace/tests/suites/0_stateless/05_ddl/05_0000_ddl_drop_tables.stdout	2022-05-26 23:50:47.354907500 +0000
> @@ -0,0 +1,3 @@
> +ERROR 1105 (HY000) at line 12: Code: 1005, displayText = sql parser error: Expected end of statement, found: ALL.
> +ERROR 1105 (HY000) at line 14: Code: 2302, displayText = Table 't' already exists.
> +ERROR 1105 (HY000) at line 15: Code: 1005, displayText = sql parser error: Expected end of statement, found: ALL.
> ```
> 
> How to deal with this case: The test file is for the new version query, and the old-version query to do the compact test? cc @drmingdrmer

Hmm...test-compat should run the test cases only valid for the old query...

Let me find a way to filter them out

**Pull request merged**